app/Http/Controllers/Dashboard/ApiController.php

```diff
@@ -38,15 +38,15 @@ public function postUpdateComponent(Component $component)
         try {
             execute(new UpdateComponentCommand(
                 $component,
-                $component->name,
-                $component->description,
+                null,
+                null,
                 Binput::get('status'),
-                $component->link,
-                $component->order,
-                $component->group_id,
-                $component->enabled,
-                $component->meta,
-                $component->tags,
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
```

Comment:
This definitely doesn't look right to me...

Reply:
Hum.. that's the same implemented anywhere else?  Example from ./app/Bus/Handlers/Commands/Incident/CreateIncidentCommandHandler.php:


```
        // Update the component.
        if ($component = Component::find($command->component_id)) {
            execute(new UpdateComponentCommand(
                Component::find($command->component_id),
                null,
                null,
                $command->component_status,
                null,
                null,
                null,
                null,
                null,
                null,
                true  // Silent mode
            ));
```

To see the problem, try updating the dashboard multiple times, and have a look to flags in DB after each changes. After 3 or 4 changes, you will have multiple error 500.

Pierre

Reply:
> To see the problem, try updating a component with no flag in the dashboard multiple times, and have a look to flags in DB after each update.

I mean: with my PR's merged last week. They broke components updates from the gui. The current PR fix them.